cmd/mdatagen/internal/templates/logs.go.tmpl

```diff
@@ -15,14 +15,38 @@ import (
 )
 
 
+{{ if .Events }}
+type EventOption interface {
+	apply(plog.LogRecord)
+}
+
+type eventOptionFunc func(plog.LogRecord)
+
+func (eof eventOptionFunc) apply(lr plog.LogRecord) {
+	eof(lr)
+}
+
+func WithEventTraceId(traceID pcommon.TraceID) EventOption {
+	return eventOptionFunc(func(lr plog.LogRecord) {
+		lr.SetTraceID(traceID)
+	})
+}
+
+func WithEventSpanId(spanID pcommon.SpanID) EventOption {
+	return eventOptionFunc(func(lr plog.LogRecord) {
+		lr.SetSpanID(spanID)
+	})
+}
+{{ end }}
+
 {{ range $name, $event := .Events -}}
 type event{{ $name.Render }} struct {
 	data     plog.LogRecordSlice // data buffer for generated log records.
 	config   EventConfig         // event config provided by user.
 }
 
 func (e *event{{ $name.Render }}) recordEvent(timestamp pcommon.Timestamp
```

Comment:
Can we accept "ctx" here instead of passing the options?

Reply:
Updated.

Reply:
By the way, I'm using `context.Context` as the parameter type since it's more commonly used.